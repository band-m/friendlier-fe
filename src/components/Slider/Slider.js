import React, { useState } from 'react';
import Nouislider from 'nouislider-react';
import 'nouislider/distribute/nouislider.css';

export default function Slider(){
  const [slider1, setSlider1] = useState(10);
  const [slider2, setSlider2] = useState(20);
  const [num, setNum] = useState(1);
  const [numOfDays, setNumOfDays] = useState(30);
  const [deadline, setDeadline] = useState('months');

  const update = (render, handle, value, un, percent) => {    
    if(handle === 0){
      setSlider1(value[0].toFixed(0))
    } else {
      setSlider2(value[1].toFixed(0))
    }
  }

  const changeNumOfDaysRadio = target => {
    setDeadline(target.id);
    switch(target.id){
      case 'days':
        return setNumOfDays(num)
      case 'weeks':
        return setNumOfDays(num * 7)
      case 'months':
        return setNumOfDays(num * 30)
    }
  }

  const changeNumOfDaysInput = value => {
    setNum(value);
    switch(deadline){
      case 'days':
        return setNumOfDays(value)
      case 'weeks':
        return setNumOfDays(value * 7)
      case 'months':
        return setNumOfDays(value * 30)
    }
  }

  return (
    <>
      <p>How often do you want to be in contact with name?</p>
      <p>Every</p>
      <input type='number' min={1} value={num} onChange={({target}) => changeNumOfDaysInput(+target.value)} /><br/>
      <label htmlFor='days'>Days</label>
      <input type='radio' id='days' name='deadline' checked={deadline === 'days'} onChange={({target}) => changeNumOfDaysRadio(target)}/>
      <label htmlFor='weeks'>Weeks</label>
      <input type='radio' id='weeks' name='deadline' checked={deadline === 'weeks'} onChange={({target}) => changeNumOfDaysRadio(target)}/>
      <label htmlFor='months'>Months</label>
      <input type='radio' id='months' name='deadline' checked={deadline === 'months'} onChange={({target}) => changeNumOfDaysRadio(target)}/><br/>

      <input style={{ marginTop: '50px' }} type='number' value={slider1} onChange={({target}) => setSlider1(target.value)} />
      <input type='number' value={slider2} onChange={({target}) => setSlider2(target.value)} />
      <input type='number' value={numOfDays} onChange={({target}) => setNumOfDays(+target.value)} />

      <Nouislider onSlide={update} range={{ min: 0, max: numOfDays }} start={[slider1, slider2]} margin={1} tooltips={[true, true]} connect step={1} pips={{ mode: 'steps' }} />
    </>
  )
}
