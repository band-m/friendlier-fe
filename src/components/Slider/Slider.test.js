import React from 'react';
import { shallow } from 'enzyme';
import Slider from './Slider';

describe('Slider component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Slider />);
    expect(wrapper).toMatchSnapshot();
  });
});
