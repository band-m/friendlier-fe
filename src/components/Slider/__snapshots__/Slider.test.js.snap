// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component matches snapshot 1`] = `
<article>
  <div>
    How often do you want to be in contact with name?
    <p>
      Every
    </p>
    <input
      min={1}
      onChange={[Function]}
      type="number"
      value={2}
    />
    <label
      htmlFor="days"
    >
      Days
    </label>
    <input
      checked={false}
      id="days"
      name="deadlineUnit"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor="weeks"
    >
      Weeks
    </label>
    <input
      checked={true}
      id="weeks"
      name="deadlineUnit"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor="months"
    >
      Months
    </label>
    <input
      checked={false}
      id="months"
      name="deadlineUnit"
      onChange={[Function]}
      type="radio"
    />
    <p>
      Choose your notification range for name:
    </p>
    <label
      htmlFor={1}
    >
      1
    </label>
    <input
      checked={true}
      id={1}
      name="notificationOptions"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor={2}
    >
      2
    </label>
    <input
      checked={false}
      id={2}
      name="notificationOptions"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor={3}
    >
      3
    </label>
    <input
      checked={false}
      id={3}
      name="notificationOptions"
      onChange={[Function]}
      type="radio"
    />
     
    <br />
     
    <br />
     
    <br />
  </div>
  <p>
    Your connection deadline with name will be on 
  </p>
  <p>
    Your yellow zone will begin on 
  </p>
  <p>
    Your red zone will begin on 
  </p>
  <Memo(Nouislider)
    connect={
      Array [
        true,
        true,
        true,
      ]
    }
    margin={1}
    onSlide={[Function]}
    pips={
      Object {
        "density": 4,
        "mode": "steps",
      }
    }
    range={
      Object {
        "max": 14,
        "min": 0,
      }
    }
    start={
      Array [
        10,
        20,
      ]
    }
    step={1}
    style={
      Object {
        "margin": "25px",
      }
    }
    tooltips={
      Array [
        true,
        true,
      ]
    }
  />
</article>
`;
